### PR TITLE
accounts hash scan refactoring

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
add comments and refactor accounts hash scanning

#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -6899,24 +6901,32 @@ impl AccountsDb {
...
         let first_chunk_start = ((first_non_ancient_slot + MAX_ITEMS_PER_CHUNK)
             / MAX_ITEMS_PER_CHUNK)
             * MAX_ITEMS_PER_CHUNK;
 
-        let non_ancient_slot_count = max_slot_inclusive - first_non_ancient_slot + 1;
+        let non_ancient_slot_count = (max_slot_inclusive - first_non_ancient_slot + 1) as usize;
+
+        // 2 is for 2 special chunks - unaligned slots at the beginning and end
+        let chunk_count =
+            ancient_slot_count + 2 + non_ancient_slot_count / (MAX_ITEMS_PER_CHUNK as usize);
```
[the remark this answers is not in the document]
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:6914]: No. Those chunks will produce empty ranges. I'm about to add a test confirming all edge cases. This refactoring is a way to make this code much more testable.

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -7002,7 +7009,7 @@ impl AccountsDb {
                     || config.store_detailed_debug_info_on_failure
                 {
                     let mut load_from_cache = true;
-                    let mut hasher = std::collections::hash_map::DefaultHasher::new(); // wrong one?
+                    let mut hasher = std::collections::hash_map::DefaultHasher::new();
```
[the remark this answers is not in the document]
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:7012]: this is ~2500 hashes. I think this will be fast enough whatever hasher we use.
